src/util/file-filter.js

```diff
@@ -0,0 +1,119 @@
+/* @flow */
+import path from 'path';
+
+import minimatch from 'minimatch';
+
+import {createLogger} from './logger';
+
+const log = createLogger(__filename);
+
+// Use this function to mimic path.resolve without resolving to absolute path.
+export const normalizeResolve = (file: string): string => {
+  // normalize
+  file = path.normalize(file);
+
+  // trim trailing slash
+  if (path.parse(file).base && file.endsWith(path.sep)) {
+    return file.slice(0, -1);
+  }
+  return file;
+};
+
+// FileFilter types and implementation.
+
+export type FileFilterOptions = {|
+  filesToIgnore?: Array<string>,
+  ignoreFiles?: Array<string>,
+  sourceDir?: string,
+  artifactsDir?: string,
+|};
+
+/*
+ * Allows or ignores files.
+ */
+export class FileFilter {
+  filesToIgnore: Array<string>;
+  sourceDir: ?string;
+
+  constructor({
+    filesToIgnore = [
```

Comment:
@eight04 yeah, I agree that not being able to use the ES6 syntax shortcut is a pity.

What I'd like to achieve is to make it clear that the FileFilter constructor property currently named `filesToIgnore` is not overridden by the new `ignoreFiles` property, but extended.

How about rename `filesToIgnore` to `baseIgnoredPatterns` (or `commonIgnoredPatterns`) and leave `ignoreFiles` unmodified?